notebooks/ch12/linear_projection.md

```diff
@@ -0,0 +1,330 @@
+
+So if we can express each $y_i - \begin{bmatrix} 1 & x_i \end{bmatrix}
+\begin{bmatrix} 
+     \hat{\theta_0} \\
+     \hat{\theta_1}
+\end{bmatrix}$ as a vector, we can express our loss as the *L2*-norm, squared:
+
+$$ 
+\begin{aligned}
+L(\hat{\theta}, \vec{x}, \vec{y})
+&= \left \Vert  \qquad   
+\begin{bmatrix} y_1 \\ y_2 \\ y_3  \end{bmatrix} \quad - \quad 
+\begin{bmatrix} 1 & x_1 \\ 1 & x_2 \\ 1 & x_3 \end{bmatrix}
+\begin{bmatrix} 
+     \hat{\theta_0} \\
+     \hat{\theta_1}
+\end{bmatrix}
+\qquad \right \Vert ^2 \\
+&= \left \Vert  \qquad  
+\vec{y} 
+\quad - \quad 
+X 
+\begin{bmatrix} 
+     \hat{\theta_0} \\
+     \hat{\theta_1}
+\end{bmatrix}
+\qquad \right \Vert ^2 \\
+&= \left \Vert  \qquad  
+\vec{y} 
+\quad - \quad 
+\vec{\hat{y}}
+\qquad \right \Vert ^2 \\
+\end{aligned}
+$$
+
+This gives us a new perspective on what it means to minimize the least squares error.
+
+
+$X$ and $\vec{y}$ are fixed, but $\hat{\theta_0}$ and $\hat{\theta_1}$ can take on any value, so there are *infinite possible vectors* created by matrix multiplying $X$ and $\hat{\theta}$. To have the smallest loss, we want to choose $\hat{\theta_0}$ and $\hat{\theta_1}$ such that the resulting vector is as close to $\vec{y}$ as possible. This is the same underlying logic as before: find the smallest squared distance; but between two vectors $\vec{y}$ and $\vec{\hat{y}}$, rather than considering the indefinite number of data points. Lastly, notice that the matrix multiplication is a linear combination of the columns of $X$: each $\hat{\theta_i}$ only ever multiplies with one column of $X$--this perspective shows us that our $\hat{y}$ is a linear combination of the features of our data. Let's get some geometric intuition on these observations.
+
+
+
+## Geometrical Intuition
+
+Multiplying $X$ and $\hat{\theta}$ represents adding scalar multiples of the columns of $X$: $\vec{x}$ and $\vec{1}$, resulting in a linear combination. Here is an intuition:
+
+Let's take two generic vectors, $\vec{x}$ and $\vec{1}$, colored black. We use red to represent the new vector we are adding, and blue for the older linear combinations.
+
+
+```python
+# HIDDEN
+plt.quiver([0, 0], [0, 0], [1, 2], [1, -4], 
+           angles='xy', scale_units='xy', scale=1, 
+           color=['black', 'black'])
+plt.xlim(-7, 7)
+plt.ylim(-8, 8)
+plt.show()
+```
+
+
+![png](linear_projection_files/linear_projection_12_0.png)
+
+
+When $\hat{\theta_0} = 2$ and $\hat{\theta_1} = 0.5$, we add half of $x$ with two times the unit vector, resulting in a new vector.
+
+
+```python
+# HIDDEN
+plt.quiver([0, 0, 0], [0, 0, 0], [1, 2, 3], [1, -4, 0], 
+           angles='xy', scale_units='xy', scale=1, 
+           color=['black', 'black', 'red'])
+plt.xlim(-7, 7)
+plt.ylim(-8, 8)
+plt.show()
+```
+
+
+![png](linear_projection_files/linear_projection_14_0.png)
+
+
+When $\hat{\theta_0} = 3$ and $\hat{\theta_1} = -1$, we add $3$ times the unit vector to $-1$ times $x$, resulting in yet another vector.
```

Comment:
in yet another vector, which is colored red below.

I think at this point they can understand that the new ones will be colored red.